fcrepo-clients/src/main/java/edu/unc/lib/dl/fcrepo4/RepositoryObjectFactory.java

```diff
@@ -307,8 +307,27 @@ public URI createMemberLink(URI parentUri, URI memberUri) throws FedoraException
 				parentUri, memberUri);
 	}
 
-	public URI createPremisEvent(URI objectUri, Model model) throws FedoraException {
-		return null;
+	/**
+	 * Creates a fedora object at the given location with the provided
+	 * properties
+	 * 
+	 * @param uri
+	 * @param model
+	 * @return
+	 * @throws FedoraException
+	 */
+	public URI createObject(URI uri, Model model) throws FedoraException {
+
+		try (FcrepoResponse response = getClient().put(uri)
+				.body(RDFModelUtil.streamModel(model), TURTLE_MIMETYPE)
+				.perform()) {
+
+			return response.getLocation();
+		} catch (IOException e) {
+			throw new FedoraException("Unable to create object at " + uri, e);
+		} catch (FcrepoOperationFailedException e) {
+			throw ClientFaultResolver.resolve(e);
```

Comment:
I believe the IOException comes from closing the response, and the FcrepoOperationFailedException is thrown by any call to the fcrepoClient.put/get/post/etc methods, and is thrown whenever a 400 or 500 status code is returned I believe.